network/crl.go

```diff
@@ -61,6 +119,11 @@ type CRLVerifier interface {
 
 type DefaultCRLVerifier struct {
 	Config CRLConfig
+	Client CRLClient
+	// key = URI of cached CRL
+	// value = parsed CRL
+	// XXX maybe hide it behind mutex?
+	cache map[string]*pkix.CertificateList
```

Comment:
IIRC, Acra *can* accept and make TLS connections from different goroutines, so it makes sense to synchronize mutable data. Go has `sync.Map` which might be useful instead of explicit locking, though.

@Lagovas, please correct me.

Reply:
Yes, at first we accept connection in main goroutine, then we pass it to new goroutine which may do tls handshake, plus will create new connection to database. And both of them will use verifiers. So yes, we should use thread-safe shared struct. And better to use sync.Map or mutex.
Plus as I told at standup, better to introduce interface for cache like 
```
type VerifierCache interface {
  Get(key []byte)([]byte, error)
  Put(key, data []byte) error
}
```
to easily change implementation in a future to some distributed storage.  Plus we should separate cache for each verifier and for each purpose: ocsp/crl. It will depends on implementation. By key prefix if it will store in redis, or separate tables/names in tables, or tree structure in boltdb with separate buckets for each purpose/direction of connections